src/view/SearchView.java

```diff
@@ -25,8 +28,12 @@ public class SearchView extends JPanel implements ActionListener, PropertyChange
     private final SearchController searchController;
     // create button for search.
     final JButton search;
+    final JButton getHistory;
+    final JButton clearHistory;
 
-    public SearchView(SearchViewModel searchViewModel, SearchController searchController) {
+
+    public SearchView(SearchViewModel searchViewModel, SearchController searchController, SaveController saveController,
+                      GetHistoryController getHistoryController, ClearController clearController) {
```

Comment:
add
this.(name of the controller) = (name of the controller);
for each controller now added as instance attributes